ktor-server/ktor-server-core/jvmAndNix/src/io/ktor/server/application/ApplicationPlugin.kt

```diff
@@ -34,17 +34,27 @@ public interface Plugin<
 }
 
 /**
- * Defines a Plugin that is installed into Application
+ * Defines a Plugin that is installed into Application.
  * @param TPipeline is the type of the pipeline this plugin is compatible with
  * @param TConfiguration is the configuration object type for this Plugin
  * @param TPlugin is the instance type of the Plugin object
  */
 @Suppress("AddVarianceModifier")
-public interface ApplicationPlugin<
+public interface BaseApplicationPlugin<
     in TPipeline : Pipeline<*, ApplicationCall>,
     out TConfiguration : Any,
     TPlugin : Any> : Plugin<TPipeline, TConfiguration, TPlugin>
 
+/**
+ * Defines a Plugin that is installed into Application
+ * @param TPipeline is the type of the pipeline this plugin is compatible with
+ * @param TConfiguration is the configuration object type for this Plugin
+ * @param TPlugin is the instance type of the Plugin object
+ */
+@Suppress("AddVarianceModifier")
+public interface ApplicationPlugin<out TConfiguration : Any> :
```

Comment:
should it be type alias or do we need a real type?

Reply:
I will check if the type alias appears in the IDE actions and if so we can use it

Reply:
Checked. Typealias are not used when you specify an explicit type. So interface should be used